core/src/main/scala/akka/persistence/r2dbc/internal/postgres/PostgresQueryDao.scala

```diff
@@ -461,27 +453,37 @@ private[r2dbc] class PostgresQueryDao(settings: R2dbcSettings, executorProvider:
   }
 
   override def persistenceIds(afterId: Option[String], limit: Long): Source[String, NotUsed] = {
-    // FIXME
-    require(settings.numberOfDataPartitions == 1, "persistenceIds not implemented for more than one data-partition yet")
-    val executor = executorProvider.executorFor(slice = 0)
-    val result = executor.select(s"select persistenceIds")(
-      connection =>
-        afterId match {
-          case Some(after) =>
-            val stmt = connection.createStatement(allPersistenceIdsAfterSql)
-            bindAllPersistenceIdsAfterSql(stmt, after, limit)
-
-          case None =>
-            connection
-              .createStatement(allPersistenceIdsSql)
-              .bind(0, limit)
-        },
-      row => row.get("persistence_id", classOf[String]))
+    val actualLimit = if (limit > Int.MaxValue) Int.MaxValue else limit.toInt
+    val results: immutable.IndexedSeq[Future[immutable.IndexedSeq[String]]] =
+      // query each data partition
+      settings.dataPartitionSliceRanges.map { sliceRange =>
+        val executor = executorProvider.executorFor(sliceRange.min)
+        executor.select(s"select persistenceIds")(
+          connection =>
+            afterId match {
+              case Some(after) =>
+                val stmt = connection.createStatement(allPersistenceIdsAfterSql(sliceRange.min))
+                bindAllPersistenceIdsAfterSql(stmt, after, actualLimit)
+
+              case None =>
+                connection
+                  .createStatement(allPersistenceIdsSql(sliceRange.min))
+                  .bind(0, actualLimit)
+            },
+          row => row.get("persistence_id", classOf[String]))
+      }
```

Comment:
Maybe the expectation is that the source is entries when `persistenceIds` call was made rather than the stream materialization, but this could be surprinsing, should probably be triggered by the futureSource rather than just collected there?

Reply:
I think that is the same for everything here, e.g. `eventsByPersistenceId`. So if it's better with lazy query we should change that in a separate pr.

Reply:
Yeah, maybe that is anyway the expectation, let's leave this and the others as is.